File: src/main/java/net/Chidoziealways/everythingjapanese/particle/ModParticles.java
package net.Chidoziealways.everythingjapanese.particle;

import net.Chidoziealways.everythingjapanese.EverythingJapanese;
import net.minecraft.core.particles.ParticleType;
import net.minecraft.core.particles.SimpleParticleType;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import net.minecraftforge.registries.RegistryObject;

public class ModParticles {
    public static final DeferredRegister<ParticleType<?>> PARTICLE_TYPES =
            DeferredRegister.create(ForgeRegistries.PARTICLE_TYPES, EverythingJapanese.MOD_ID);

    public static final RegistryObject<SimpleParticleType> PYRITE_PARTICLES =
            PARTICLE_TYPES.register("pyrite_particles", () -> new SimpleParticleType(true));

    public static void register(IEventBus eventBus) {
        PARTICLE_TYPES.register(eventBus);
    }
}

File: src/main/kotlin/net/Chidoziealways/everythingjapanese/particle/ModParticles.kt
package net.Chidoziealways.everythingjapanese.particle

import net.Chidoziealways.everythingjapanese.JAPANESE_MOD_ID
import net.minecraft.core.particles.SimpleParticleType
import net.minecraft.core.registries.Registries
import net.neoforged.bus.api.IEventBus
import net.neoforged.neoforge.registries.DeferredRegister
import thedarkcolour.kotlinforforge.neoforge.forge.getValue
import java.util.function.Supplier

object ModParticles {
    val PARTICLE_TYPES = DeferredRegister.create(Registries.PARTICLE_TYPE, JAPANESE_MOD_ID)

    val PYRITE_PARTICLES by PARTICLE_TYPES.register("pyrite_particles", Supplier { SimpleParticleType(true) })

    fun register(eventBus: IEventBus) {
        PARTICLE_TYPES.register(eventBus)
    }
}
